homeassistant/components/soundtouch/media_player.py

```diff
@@ -357,6 +373,17 @@ def play_media(self, media_type, media_id, **kwargs):
             else:
                 _LOGGER.warning("Unable to find preset with id %s", media_id)
 
+    def select_source(self, source):
+        """Select input source."""
+        if source == Source.AUX.value:
+            _LOGGER.debug("Selecting source AUX")
+            self._device.select_source_aux()
```

Comment:
Here I think it would make sense to have a `select_source(Source)` method in the upstream lib and use that instead of having separate methods per source type.

Reply:
Indeed, but the upstream lib isn't maintained anymore, so we're stuck for now.
There are efforts to replace the library, but some features are missing.
https://github.com/CharlesBlonde/libsoundtouch/issues/38